errors/errors.go

```diff
@@ -0,0 +1,62 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"runtime"
+)
+
+type Error struct {
+	function FCall
+	Err      error
+	Severity logrus.Level
+}
+func (e Error) Error() string {
+	return e.Err.Error()
+}
+type FCall string
+
+func (o FCall) String() string {
+	return string(o)
+}
+
+func E(functionCall FCall, args ...interface{}) error {
+	e := Error{function: functionCall}
+	if len(args) == 0 {
+		msg := "errors.E called with 0 args"
+		_, file, line, ok := runtime.Caller(1)
+		if ok {
+			msg = fmt.Sprintf("%v - %v:%v", msg, file, line)
+		}
+		e.Err = errors.New(msg)
+	}
+	for _, a := range args {
+		switch a := a.(type) {
+		case error:
+			e.Err = a
+		case string:
+			e.Err = errors.New(a)
```

Comment:
When providing a err and string as in line 200, will it not just overwrite e.Err with the last one.

Reply:
Actually, it will act according to type of given argument, so I do not see any overwrite situation here. For instance, it will be used as following... 
`daemon.go`

```
import ( 
   cError "github.com/aau-network-security/haaukins/errors"
)
func New(....) error {
const fCall cError.FCall = "daemon.New"

if err!= .... ;  err!=nil {
    return cError.E(fCall,err)
}
// or in case of we have defined constant error variables
if err!= .... ;  err!=nil {
    return cError.E(fCall,UnknownMethodErr)
}
}
```
I hope, I understood your statement.

Reply:
I don't think i understand it correctly then. What happens if one provided a string and err like this

```
cError.E(op, err, fmt.Sprintf("unable to read users file: %s", conf.UsersFile))
```

Wouldn't the string just replace the err?

Reply:
I don't think i understand it correctly then. What happens if one provided a string and err like this

```
cError.E(op, err, fmt.Sprintf("unable to read users file: %s", conf.UsersFile))
```

Wouldn't the string just replace the err?

Reply:
Yeah, true in that case it will overwrite, but can't it be written as following ?  ; 
```
cError.E(op,fmt.Sprintf("unable to read users file: %s err: %s", conf.UsersFile,err))
```

Reply:
Yes but why not have it function like https://godoc.org/github.com/pkg/errors#Wrapf.
So if you apply a string and err it will look something like this.

```
Applied string: applied error
```

Reply:
so, do you think that providing Fcall into E is not useful at all ? or we do not need such a thing and just use wrapf ?

Reply:
No the fcall is a good idea. But I think we should make it so that it accepts both err and strings and then combines them like wrapf